lib/ansible/vars/manager.py

```diff
@@ -196,16 +196,11 @@ def _combine_and_track(data, new_data, source):
         if self.safe_basedir:  # avoid adhoc/console loading cwd
             basedirs = [self._loader.get_basedir()]
 
-        if play:
+        if play and host:
             for role in play.get_roles():
-                # role is public and
-                #    either static or dynamic and completed
-                # role is not set
-                #    use config option as default
-                role_is_static_or_completed = role.static or role._completed.get(host.name, False)
-                if role.public and role_is_static_or_completed or \
-                   role.public is None and not C.DEFAULT_PRIVATE_ROLE_VARS and role_is_static_or_completed:
-                    all_vars = _combine_and_track(all_vars, role.get_default_vars(), "role '%s' defaults" % role.name)
+                if role.static or role._completed.get(host.name, False):
+                    all_vars = _combine_and_track(all_vars, role.get_default_vars(), f"role '{role.name}' defaults")
```

Comment:
A similar change can be made here:

```suggestion
        for role in play.get_roles() if play and host else ():
            if role.static or role._completed.get(host.name, False):
                all_vars = _combine_and_track(all_vars, role.get_default_vars(), f"role '{role.name}' defaults")
```

Reply:
no, `play and host` would be incorrect as role vars should be available even w/o host, you could make 'non static' imports only count once you have host, but even then, once completed they should still be usable as play vars.